test/onnx/test_pytorch_onnx_onnxruntime.py

```diff
@@ -6842,6 +6851,124 @@ def forward(self, x):
+        model.eval()
+        x = torch.ones(2, 3, 4, 5, 6)
+        self.run_test(model, x)
+
+    @skipIfUnsupportedMinOpsetVersion(9)
+    def test_symbolic_shape_inference_arange(self):
+        # test Range
+        class ArangeModel(torch.nn.Module):
+            def forward(self, signal):
+                frame_step = 2
+                outer_dimensions = signal.size()[:-2]
+                frames, frame_length = signal.size()[-2:]
+
+                subframe_length = signal.size()[0]
+                subframe_step = frame_step // subframe_length
+                subframes_per_frame = frame_length // subframe_length
+                output_size = frame_step * (frames - 1) + frame_length
+                output_subframes = output_size // subframe_length
+
+                frame = torch.arange(0, output_subframes)
+                return frame
+
+        model = ArangeModel()
+        model.eval()
+        M, C, K, N = 1, 2, 3, 4
+        x = torch.randint(5, (M, C, K, N))
+        y = torch.randint(5, (M, C + 1, K + 1, N + 1))
+        self.run_test(model, x)
+        self.run_test(model, x, input_names=['x'],
+                      dynamic_axes={'x' : [0, 1, 2, 3]}, test_with_inputs=[(x,), (y,)])
+
+    @skipIfUnsupportedMinOpsetVersion(11)
+    def test_symbolic_shape_inference_box(self):
+        # test NonZero
+        class BoxModel(torch.nn.Module):
+            def forward(self, boxes):
+                min_size = 1e-2
+                ws, hs = boxes[:, 2] - boxes[:, 0], boxes[:, 3] - boxes[:, 1]
+                keep = (ws >= min_size) & (hs >= min_size)
+                keep = torch.where(keep)[0]
+                return keep
+
+        model = BoxModel()
+        model.eval()
+        x = torch.ones(2, 4)
+        y = torch.ones(3, 5)
+        self.run_test(model, x)
+        self.run_test(model, x, input_names=['x'],
+                      dynamic_axes={'x' : [0, 1]}, test_with_inputs=[(x,), (y,)])
+
+    @skipIfUnsupportedMinOpsetVersion(11)
+    def test_symbolic_shape_inference_box_if(self):
+        # test If
+        class BoxIfModel(torch.nn.Module):
+            def forward(self, boxes, scores):
+                score_thresh = 0.0
+                inds = torch.where(scores > score_thresh)[0]
+                boxes_1 = boxes[inds]
+                if boxes_1.numel() > 3:
+                    return boxes_1
+                else:
+                    return boxes_1 * 2
+
+        model = BoxIfModel()
+        model.eval()
+        boxes = torch.ones(2, 4)
+        scores = torch.ones(1, 4)
+        self.run_test(model, (boxes, scores))
+
+    @skipIfUnsupportedMinOpsetVersion(11)
+    def test_symbolic_shape_inference_arange_2(self):
+        # test Range
+        class ArangeModel(torch.nn.Module):
+            def forward(self, start):
+                return torch.arange(start.size(0), 8.5, 1.5, dtype=torch.int64)
+        x = torch.randn(2, 3, 4)
+        y = torch.tensor(8.5, dtype=torch.float)
```

Comment:
nit: y is not needed here, it's not used

Reply:
done